Source/src/components/ComponentTransform.h

```diff
@@ -38,120 +38,53 @@ namespace Hachiko
 
-        static Type GetType();
-        [[nodiscard]] bool HasChanged() const;
-        [[nodiscard]] const float3& GetPosition() const;
-        [[nodiscard]] const float3& GetLocalPosition() const;
-        [[nodiscard]] const float3& GetScale() const;
-        [[nodiscard]] const float3& GetLocalScale() const;
-        [[nodiscard]] const Quat& GetRotation() const;
-        [[nodiscard]] const Quat& GetLocalRotation() const;
-        [[nodiscard]] const float3& GetRotationInEulerAngles() const;
-        [[nodiscard]] const float3& GetLocalRotationInEulerAngles() const;
-        [[nodiscard]] const float3& GetFront() const;
-        [[nodiscard]] const float3& GetUp() const;
-        [[nodiscard]] const float3& GetRight() const;
-        [[nodiscard]] const float4x4& GetMatrix() const;
-        [[nodiscard]] const float4x4& GetLocalMatrix() const;
-       
+        static Type GetType() { return Type::TRANSFORM; };
+        [[nodiscard]] bool IsDirty() const { return dirty; }
+        [[nodiscard]] bool HasChanged() const { return changed; }
+        [[nodiscard]] const float3& GetLocalPosition() const { return position_local; }
+        [[nodiscard]] const float3& GetLocalScale() const { return scale_local; }
+        [[nodiscard]] const Quat& GetLocalRotation() const { return rotation_local; }
+        [[nodiscard]] const float3& GetLocalRotationInEulerAngles() const { return rotation_euler_local; }
+        [[nodiscard]] const float4x4& GetLocalMatrix() const { return matrix_local; }
+
+        [[nodiscard]] const float3& GetPosition();
+        [[nodiscard]] const float3& GetScale();
+        [[nodiscard]] const Quat& GetRotation();
+        [[nodiscard]] const float3& GetRotationInEulerAngles();
+        [[nodiscard]] const float3& GetFront();
+        [[nodiscard]] const float3& GetUp();
+        [[nodiscard]] const float3& GetRight();
+        [[nodiscard]] const float4x4& GetMatrix();
+
+
         void Save(JsonFormatterValue j_component) const override;
         void Load(JsonFormatterValue j_component) override;
 
         void DrawGui() override;
 
     private:
         void CalculateTransform(MatrixCalculationMode calculation_mode);
-        void UpdateTransformOfHierarchy(MatrixCalculationMode matrix_calculation_mode);
+        void UpdateTransformAndChildren(MatrixCalculationMode matrix_calculation_mode);
+        void Invalidate();
 
     private:
-        bool        changed = true;
-        float4x4    matrix;
-        float4x4    matrix_local;
-        float3      position;
-        float3      position_local;
-        float3      scale;
-        float3      scale_local;
-        Quat        rotation;
-        Quat        rotation_local;
-        float3      rotation_euler;
-        float3      rotation_euler_local;
-        float3      right;
-        float3      up;
-        float3      front;
+        bool dirty = true;          // Symbolizes whether the GLOBAL transform needs to be re calculated or not.
```

Comment:
```suggestion
        bool dirty;
```
Since we already initialize it in the constructor.

Reply:
Same with the changed flag